conda/cli/main_env_create.py

```diff
@@ -141,70 +154,97 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
     prefix_data.validate_path()
     prefix_data.validate_name()
 
-    # TODO, add capability
-    # common.ensure_override_channels_requires_channel(args)
-    # channel_urls = args.channel or ()
-
-    result = {"conda": None, "pip": None}
+    # Convert all environment configuration to environment variables for fresh_context
+    # NOTE: Only channels and environment variables are currently supported in
+    # environment.yaml files.
+    env_vars = {}
+    # Add environment variables from YAML (convert all values to strings)
+    if env.variables:
+        env_vars.update({k: str(v) for k, v in env.variables.items()})
+
+    if env.config.channels:
+        env_vars["CONDA_CHANNELS"] = ",".join(env.config.channels)
+
+    # TODO: Add other config options when they're supported in environment.yaml
+    # if env.config.channel_priority:
+    #     env_vars['CONDA_CHANNEL_PRIORITY'] = str(env.config.channel_priority)
+    # if env.config.solver:
+    #     env_vars['CONDA_SOLVER'] = env.config.solver
+
+    # Use fresh_context to load all environment configuration without writing to disk
+    with fresh_context(argparse_args=args, env=env_vars):
```

Comment:
Thats a neat idea! Taking this approach, we don't need to change conda's behaviour when it comes to writing an environment level condarc when an environment is created. @jezdez @kenodegard what do you think?